verilog/tools/ls/verilog-language-server.cc

```diff
@@ -17,14 +17,20 @@
 #include <functional>
 #include <memory>
 
+#include "absl/flags/flag.h"
 #include "absl/strings/string_view.h"
 #include "common/lsp/lsp-protocol.h"
 #include "common/util/file_util.h"
 #include "common/util/init_command_line.h"
+#include "verilog/tools/ls/lsp-file-utils.h"
 #include "verilog/tools/ls/verible-lsp-adapter.h"
 
 namespace verilog {
 
+void VerilogLanguageServer::UpdateFlags() {
+  absl::SetFlag(&FLAGS_rules_config_search, true);
```

Comment:
I am not convinced actually that we should do the upward search. It feels 'too magic'. And it also can fail in surprising ways in various occasions (e.g. if people have constructed their projects with symbolic links).

So for the language server I would suggest to not use the automatic searching, but look at the exact same directory  where the project root is. Either the project root configured coming from the editor or the project root established by finding `verible.filelist`.

Reply:
I disabled automatic searching (I left a flag declaration, because I use it in LS tests).

In my opinion, having two mechanisms searching for lint configuration (enabling it with `--rules-config-search` and additional check in project root) may be also prone to errors, maybe we should highlight in README that it should be enabled in the configuration for launching this language server?

Reply:
Yes, having  multiple mechanisms will be problematic.

Anyway, I think it is good for now to merge, we can refine the behavior as we go.